portals/bring-your-own-code-samples/react-sample/src/components/AuthButton.tsx

```diff
@@ -0,0 +1,49 @@
+import { IconButton, Tooltip } from '@mui/material';
+import {
+    Login,
+    Logout
+} from '@mui/icons-material';
+import React from 'react';
+export const AuthButton = () => {
+    const username = (window as any)["Microsoft"]?.Dynamic365?.Portal?.User?.userName ?? "";
+    const firstName = (window as any)["Microsoft"]?.Dynamic365?.Portal?.User?.firstName ?? "";
+    const lastName = (window as any)["Microsoft"]?.Dynamic365?.Portal?.User?.lastName ?? "";
+    const isAuthenticated = username !== "";
+    const [token, setToken] = React.useState<string>("");
+
+    React.useEffect(() => {
+        const getToken = async () => {
+            try {
+                const token = await (window as any).shell.getTokenDeferred();
+                setToken(token);
+            } catch (error) {
+                console.error('Error fetching token:', error);
+            }
+        };
+        getToken();
+    }, []);
+
+    return (
+        <div className="flex items-center gap-4">
+            {isAuthenticated ? (
+                <>
+                    <span className="text-sm">Welcome {firstName + " " + lastName}</span>
+                    <Tooltip title="Logout">
+                        <IconButton color="primary" onClick={() => window.location.href = "/Account/Login/LogOff?returnUrl=%2F"}>
+                            <Logout />
+                        </IconButton>
+                    </Tooltip>
+                </>
+            ) : (
+                <form action="/Account/Login/ExternalLogin" method="post">
+                    <input name="__RequestVerificationToken" type="hidden" value={token} />
+                    <Tooltip title="Login">
+                        <IconButton name="provider" type="submit" color="primary" value="https://login.windows.net/<Your Tenant Id>/">
```

Comment:
Why is this hard coded? Why isn't this set using a variable in a .env file or similar?
See https://github.com/microsoft/PowerApps-Samples/blob/master/dataverse/webapi/JS/quickspa/.env.example 
and
```js
import 'dotenv/config'

// Load the environment variables from the .env file
const config = {
   baseUrl: process.env.BASE_URL,
   clientId: process.env.CLIENT_ID,
   tenantId: process.env.TENANT_ID,
   redirectUri: process.env.REDIRECT_URI,
};
```